crates/node/src/lib.rs

```diff
@@ -540,6 +551,43 @@ impl Node {
         Ok(outcome)
     }
 
+    async fn substitute_aliases_in_payload(
+        &self,
+        context_id: ContextId,
+        payload: Vec<u8>,
+        aliases: &[Alias<PublicKey>],
+    ) -> Result<Vec<u8>, CallError> {
+        if aliases.is_empty() {
+            return Ok(payload);
+        }
+
+        let mut result = Vec::with_capacity(payload.len());
+        let mut remaining = &payload[..];
+
+        for alias in aliases {
+            let placeholder = format!("{{{}}}", alias.as_ref());
+            let placeholder_bytes = placeholder.as_bytes();
```

Comment:
no need for the explicit `.as_ref()`, since `Alias<T>: Display`

```suggestion
            let needle = format!("{{{alias}}}").as_bytes();
```

Reply:
I still have to implement it as 
```rust
let needle_str = format!("{{{alias}}}");
let needle = needle_str.as_bytes();
 ```

Reply:
Lifetime issue I'm guessing, feel free to use `into_bytes`